enhancements/machine-api/nutanix-multi-nics.md

```diff
@@ -0,0 +1,128 @@
+---
+title: nutanix-multi-nics
+authors:
+reviewers:
+approvers:
+api-approvers:
+creation-date: 2024-11-05
+last-updated: 2024-11-05
+tracking-link:
+  - https://issues.redhat.com/browse/CORS-3741
+---
+
+# Nutanix: Multi-NICs for OCP Cluster Nodes
+
+## Summary
+
+Ability to install OpenShift on Nutanix with nodes having multiple NICs (multiple subnets) from IPI and for autoscaling with MachineSets.
+
+## Motivation
+
+Requested by customers:
+- Everest Digital
+- Unacle B.V
+
+### Goals
+
+- Allow users to configure multiple subnets for Nutanix pltform in the install-config.yaml file at cluster installation using IPI or UPI.
+- Allow users to configure multiple subnets via Machine/MachineSet CRs' Nutanix providerSpec to add/scale worker nodes.
+- Allow smooth cluster upgrade from older OCP versions. 
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
+As an OpenShift user, I wish to deploy clusters that allow infrastructure and worker nodes with multi-NICs support. This may be to support secondary storage networking, such as Nutanix CSI, or to support other applications with segmented network requirements.
+
+### API Extensions
+
+Currently, the “subnets” fields in both Machine/MachineSet’s Nutanix providerSpec and Nutanix FailureDomain are already array type. The only change for the api is to relax the validation rule for the “subnets” fields to allow multiple values and to ensure no duplication values are configured.
+
+We will add a featue gate "NutanixMultiSubnets" (DevPreviewNoUpgrade, TechPreviewNoUpgrade) for this feature. After QE testing complete, we will add the feature gate to the "Default" feature set.
+
+```go
+// NutanixPlatformSpec holds the desired state of the Nutanix infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type NutanixPlatformSpec struct {
+  ...
+
+  // failureDomains configures failure domains information for the Nutanix platform.
+  // When set, the failure domains defined here may be used to spread Machines across
+  // prism element clusters to improve fault tolerance of the cluster.
+  // +openshift:validation:FeatureGateAwareMaxItems:featureGate=NutanixMultiSubnets,maxItems=32
```

Comment:
Testing this change should be mentioned as part of the testing and graduation criteria please

Reply:
Mentioned in the Test Plan.